docs/static/breaking-changes.asciidoc

```diff
@@ -3,54 +3,82 @@
 other components of the Elastic stack are installed on the same instance. Also, this provides a foundation
 for future change which will allow Elastic Stack packs to be installed via this script.
 
-**Kafka Input/Output Configuration Changes:** This release added support for the new 0.9 consumer/producer API which supports security features introduced by Kafka.
+[float]
+=== Breaking Changes in Plugins
+
+* **Elasticsearch Output:** The index template for 5.0 has been changed to reflect Elasticsearch's https://www.elastic.co/guide/en/elasticsearch/reference/5.0/breaking_50_mapping_changes.html[mapping changes]. 
+Most importantly, the subfield for string multi-fields has changed from `.raw` to `.keyword` to match ES's default behavior. 
+The impact of this change to various user groups are detailed below:
+
+* New Logstash 5.0 and Elasticsearch 5.0 users - uses `keyword` from the outset for subfields. In Kibana, you would be 
+using field.keyword to perform aggregations.
+* Existing users with custom templates - most of these won't be impacted since you use a custom template.
+* Existing users with default template: Logstash does not force upgrade templates if it already exists. When you 
+intend to move to this new template and want to use `.keyword` for all string fields, you'll have to reindex existing 
+data. Elasticsearch's reindexing API can help move your data from `.raw` subfield to `.keyword`.
+
+* **Kafka Input/Output Configuration Changes:** This release added support for the new 0.10 consumer/producer API which supports security features introduced by Kafka.
 A few Configuration options were renamed to make it consistent with Kafka consumer and producer settings.
 Also, this plugin version will not work with Kafka 0.8 broker.
 
 Please see the following specific plugin documentation for new configuration options:
 
-* <<plugins-inputs-kafka,Kafka Input>>
+* <<plugins-inputs-kafka, Kafka Input>>
 * <<plugins-outputs-kafka, Kafka Output>>
 
-**Ruby Filter and Custom Plugin Developers:** With the migration to the Java Event (https://github.com/elastic/logstash/issues/4191[Issue 4191]), we have changed
-how you can access internal data. The Event object no longer returns a reference to the data. Instead, it returns a
-copy. This might change how you do manipulation of your data, especially when working with nested hashes.
-When working with nested hashes, it’s recommended that you use the `fieldref` syntax instead of using multiple brackets.
-Also note that we have introduced new Getter/Setter APIs for accessing information in the Event object.
+* **File Input:** SinceDB file is now saved in `path.data` location, not user's home. If you have manually specified `sincedb_path` 
+configuration, this change will not affect you. If you are moving from 2.x to 5.x, and would like to use the existing SinceDB file, it 
+has to be copied over to `path.data` manually to use the save state.
```

Comment:
It's more than that. It's a subdirectory of `path.data`.  It's "`path.data`/plugins/inputs/file"

See https://github.com/logstash-plugins/logstash-input-file/blob/master/lib/logstash/inputs/file.rb#L200